hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/RegionStates.java

```diff
@@ -581,9 +584,11 @@ public ServerName getRegionServerOfRegion(RegionInfo regionInfo) {
       for (ServerName serverName : onlineServers) {
         ServerStateNode serverNode = serverMap.get(serverName);
         if (serverNode != null) {
-          ensemble.put(serverNode.getServerName(), serverNode.getRegionInfoList().stream()
-            .filter(region -> !isTableDisabled(tableStateManager, region.getTable()))
-            .collect(Collectors.toList()));
+          ensemble.put(serverNode.getServerName(),
+            serverNode.getRegionInfoList().stream()
+                .filter(region -> !isTableDisabled(tableStateManager, region.getTable())
```

Comment:
filter twice is easy to read?
.filter(region -> !isTableDisabled(tableStateManager, region.getTable()).filter(region -> !region.isSplitParent())

Reply:
ok , right